analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/HashesShouldHaveUnpredictableSalt.cs

```diff
@@ -26,5 +26,27 @@ public sealed class HashesShouldHaveUnpredictableSalt : HashesShouldHaveUnpredic
 
     protected override DiagnosticDescriptor Rule => S2053;
 
-    public override bool ShouldExecute() => true;
+    public override bool ShouldExecute()
+    {
+        var walker = new VulnerableTypeWalker();
+        walker.SafeVisit(Node);
+        return walker.Result;
+    }
+
+    private sealed class VulnerableTypeWalker : SafeCSharpSyntaxWalker
+    {
+        public bool Result { get; private set; }
+
+        public override void Visit(SyntaxNode node)
+        {
+            if (!Result && node is IdentifierNameSyntax identifier)
+            {
+                Result = VulnerableTypes.Any(x => identifier.NameIs(x.TypeName));
```

Comment:
Will this work if there's a class `ChildPasswordDeriveBytes : PasswordDeriveBytes`?
I saw that it's not sealed.

Is there a UT for this?